packages/plugin-ext/src/plugin/node/debug/debug.ts

```diff
@@ -182,8 +182,13 @@ export class DebugExtImpl implements DebugExt {
     }
 
     async $createDebugSession(debugConfiguration: theia.DebugConfiguration): Promise<string> {
-        const executable = await this.getExecutable(debugConfiguration);
-        const communicationProvider = startDebugAdapter(executable);
+        let communicationProvider: CommunicationProvider;
+        if ('debugServer' in debugConfiguration) {
+            communicationProvider = connectDebugAdapter(debugConfiguration.debugServer);
```

Comment:
debugServer is just a port ? not a hostname and a port ?

Reply:
just a port

Reply:
At least I don't know cases where it should be `hostname:port`

Reply:
yes, in `vscode` it is just a port
https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/debug/node/debugger.ts#L272

Reply:
@tolusha yes thx, saw it there as well: https://code.visualstudio.com/docs/editor/debugging

Reply:
@tolusha thanks